ipa-core/src/protocol/hybrid/oprf.rs

```diff
@@ -74,17 +78,82 @@ pub const PRF_CHUNK: usize = 16;
 // multiplications per batch
 const CONV_PROOF_CHUNK: usize = 256;
 
-#[allow(dead_code)]
+#[derive(Default, Debug, Clone)]
+pub struct PRFIndistinguishableHybridReport<BK: SharedValue, V: SharedValue> {
+    prf_of_match_key: u64,
+    value: Replicated<V>,
+    breakdown_key: Replicated<BK>,
+}
+
+impl<BK, V> PRFIndistinguishableHybridReport<BK, V>
+where
+    BK: SharedValue,
+    V: SharedValue,
+{
+    /// ## Panics
+    /// it doesn't. `ShardIndex` is a u32, expanded into a u64 for the mod operation
+    /// `prf_of_match_key % shard_count` will always fit into a `ShardIndex`
+    #[must_use]
+    pub fn shard_picker(&self, shard_count: ShardIndex) -> ShardIndex {
+        let shard_count = u64::from(shard_count);
+        ShardIndex::try_from(self.prf_of_match_key % shard_count)
+            .expect("Modulo a u32 will fit in a u32")
+    }
+}
+
+impl Serializable for PRFIndistinguishableHybridReport<BA8, BA3> {
+    type Size = U12;
```

Comment:
12 bytes to store 11 bits seems a bit excessive

Reply:
BA3 and BA8 are both 2 bytes, it seems

Reply:
maybe I'm being dense here, but if the widths are
```
    const PRF_MK_SZ: usize = 8;
    const V_SZ: usize = <Replicated<BA3> as Serializable>::Size::USIZE;
    const BK_SZ: usize = <Replicated<BA8> as Serializable>::Size::USIZE;
```

then
```
        self.breakdown_key.serialize(GenericArray::from_mut_slice(
            &mut buf[Self::PRF_MK_SZ + Self::V_SZ..Self::PRF_MK_SZ + Self::V_SZ + Self::BK_SZ],
        ));
        ```

Reply:
Erik was right, I was wrong here

Reply:
there's a first for everything!

Reply:
ah, not the first time for sure